cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/s3/S3StorageClient.java

```diff
@@ -55,4 +58,26 @@ public S3Client getNativeClient() {
   public StorageType.Type getStorageType() {
     return S3_TYPE;
   }
+
+  /**
+   * Checks if the blob/object exists on the backend storage. The path is the absolute path to the
+   * object including scheme (s3://)
+   *
+   * @param path absolute path to a file including scheme
+   * @return true if path exists else false
+   */
+  @Override
+  public boolean fileExists(String path) {
+    try {
+      HeadObjectRequest headObjectRequest =
+          HeadObjectRequest.builder().bucket(getRootPrefix()).key(path).build();
```

Comment:
Did you check the headObject API spec that passing the full path including scheme and bucket name as key will work? It usually expects object name without the bucket name etc.

Reply:
You are correct. Updated to parse bucket and key from path.